sdlib/d/gc/slab.d

```diff
@@ -190,15 +239,15 @@ public:
 		// On a big-endian machine, the unused high 16 bits of a pointer will be
 		// found at the start, rather than end, of the memory that it occupies,
 		// and the freespace (newMetadata) will collide with the finalizer.
-		*finalizerPtr = newMetadata | finalizerValue;
-		e.enableMetadata(index);
 		_hasMetadata = true;
+		slotMetadata.data.finalizerData = newMetadata | finalizerValue;
+		e.enableMetadata(index);
```

Comment:
No reason to reorder here.

Reply:
slotMetadata asserts that `_hasMetadata` is true. I could also create a version that doesn't assert for internal use I guess.

But that is the reason for the reordering.

Reply:
fixed